src/components/composer/composer/composer_store.ts

```diff
@@ -677,6 +678,39 @@ export class ComposerStore extends SpreadsheetStore {
     this.colorIndexByRange = colorsToKeep;
   }
 
+  private addHeadersForSpreadingFormula(content: string) {
+    if (!content.startsWith("=")) {
+      return;
+    }
+
+    try {
+      const evaluated = this.getters.evaluateFormula(this.sheetId, content);
+      if (!isMatrix(evaluated)) {
+        return;
+      }
+
+      const rowsInEvaluated = evaluated[0].length;
+      const columnsInEvaluated = evaluated.length;
+
+      for (const dim of ["ROW", "COL"] as const) {
+        const numberOfElements = this.getters.getNumberHeaders(this.sheetId, dim);
+        const elementsInEvaluated = dim === "ROW" ? rowsInEvaluated : columnsInEvaluated;
+        const startIndex = dim === "ROW" ? this.row : this.col;
+        const margin = dim === "ROW" ? 50 : 20;
+        const missingElements = startIndex + elementsInEvaluated - numberOfElements;
```

Comment:
I believe the benefits of this for loop are not worth it.
Yes it reduces the number of lines of code, but it introduces 3 ternary branches (more cyclomatic complexity) and less explicit (more generic) variable names (`missingElements` instead of `missingRows` or `missingCols` which are far easier to understand while reading).
I think duplicating the few lines of code for rows/cols would lead to a more readable
/simpler code.